plugin/src/main/java/org/opensearch/ml/action/undeploy/TransportUndeployModelAction.java

```diff
@@ -198,10 +207,10 @@ private MLUndeployModelNodeResponse createUndeployModelNodeResponse(MLUndeployMo
         String[] modelIds = MLUndeployModelNodesRequest.getModelIds();
 
         Map<String, Integer> modelWorkerNodeCounts = new HashMap<>();
-        boolean specifiedModelIds = modelIds != null && modelIds.length > 0;
-        String[] removedModelIds = specifiedModelIds ? modelIds : mlModelManager.getAllModelIds();
-        if (removedModelIds != null) {
-            for (String modelId : removedModelIds) {
+
+
+        if (modelIds != null) {
```

Comment:
This change doesn't seem correct, shouldn't we add security check here?

Reply:
The check is being done in TransportUndeployModelsAction class